src/engine/Operation.cpp

```diff
@@ -122,12 +122,12 @@ shared_ptr<const ResultTable> Operation::getResult(bool isRoot,
               }
             });
     auto computeLambda = [this, &timer] {
-      checkCancellation([this]() { return "Before " + getDescriptor(); });
+      checkCancellation();
       runtimeInfo().status_ = RuntimeInformation::Status::inProgress;
       signalQueryUpdate();
       ResultTable result = computeResult();
 
-      checkCancellation([this]() { return "After " + getDescriptor(); });
+      checkCancellation();
```

Comment:
For this position I would like to keep the descriptor in the message, as this is generic code from the base class which currently allows us to identify operations that have no cancellation checks whatsoever internally. This power would be lost with this change. 
And an overload that takes a single string or a single lambda followed by the source_location is eaisly implemented.